File: app/src/main/java/com/twilio/video/app/ui/splash/SplashActivity.java
package com.twilio.video.app.ui.splash;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;

import com.twilio.video.app.auth.Authenticator;
import com.twilio.video.app.base.BaseActivity;
import com.twilio.video.app.ui.room.RoomActivity;

import javax.inject.Inject;

import dagger.android.AndroidInjection;

public class SplashActivity extends BaseActivity {
    @Inject Authenticator authenticator;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Intent newIntent = authenticator.loggedIn() ?
            (new Intent(this, RoomActivity.class)) :
            (new Intent(this, authenticator.getLoginActivity()));
        startActivity(newIntent);
        finish();
    }

}

File: app/src/main/java/com/twilio/video/app/ui/splash/SplashActivity.kt
/*
 * Copyright (C) 2019 Twilio, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.twilio.video.app.ui.splash

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.twilio.video.app.auth.Authenticator
import com.twilio.video.app.ui.ScreenSelector
import com.twilio.video.app.ui.room.RoomActivity
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@SuppressLint("CustomSplashScreen")
@AndroidEntryPoint
class SplashActivity : AppCompatActivity() {

    @Inject lateinit var authenticator: Authenticator
    @Inject lateinit var screenSelector: ScreenSelector

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val newIntent = if (authenticator.loggedIn())
            Intent(this, RoomActivity::class.java)
        else
            Intent(this, screenSelector.loginScreen)
        startActivity(newIntent.apply { data = intent.data })
        finish()
    }
}
